platform/osx/detect.py

```diff
@@ -36,36 +33,37 @@ def get_flags():
 
 def configure(env):
 
-    env.Append(CPPPATH=['#platform/osx'])
-
-    if (env["bits"] == "default"):
-        env["bits"] = "32"
+	## Build type
 
     if (env["target"] == "release"):
-
-        env.Append(CCFLAGS=['-O2', '-ffast-math', '-fomit-frame-pointer', '-ftree-vectorize', '-msse2'])
+        env.Prepend(CCFLAGS=['-O2', '-ffast-math', '-fomit-frame-pointer', '-ftree-vectorize', '-msse2'])
 
     elif (env["target"] == "release_debug"):
-
-        env.Append(CCFLAGS=['-O2', '-DDEBUG_ENABLED'])
+        env.Prepend(CCFLAGS=['-O2', '-DDEBUG_ENABLED'])
 
     elif (env["target"] == "debug"):
```

Comment:
"debug" is already the default value in `SConstruct`, so all platforms assume that it can be used as the fallback already.

The part which could be buggy is that there is no verification made of the input, so if you define `target=toto`, it would bypass this logic. But such check should likely go in SConstruct too.